### DISCUSSION TITLE
Error when Building Release 64

### DISCUSSION THREAD

**Discussion by B4rr3l-Rid3r** [GPUOpen-Drivers/AMDVLK#332] [Q&A]
So far was everything going great but I'm getting this error when trying to build the Release64 with:  cmake --build builds/Release64. any help will be welcome, thanks.

"vulkandriver/drivers/gpurt/src/gpurtDevice.cpp:64:10: fatal error: pipelines/g_internal_shaders.h: No such file or directory
   64 | #include <pipelines/g_internal_shaders.h>"

**Comment by Flakebi** [comment]
Hi, you need to install the directx shader compiler and glslang, see also #293.
Sorry, the error message is not great.

**Reply by B4rr3l-Rid3r** [reply]
Thank you for the repply, I dis install them with the latest version per instruction below and all seems to be going well and successful, but them I got a problem with this "pipelines/g_internal_shaders.h" which I couldn´t find on any directory.

#!/bin/bash

if [ ! -d DirectXShaderCompiler ]; then
git clone --depth=1 -b release-1.7.2212 https://github.com/microsoft/DirectXShaderCompiler.git
fi

if [ ! -d glslang ]; then
git clone --depth=1 -b sdk-1.3.250 https://github.com/KhronosGroup/glslang.git
fi

cd DirectXShaderCompiler
git submodule init
git submodule update
cmake -H. -Bbuilds -GNinja -DCMAKE_BUILD_TYPE=Release -C ./cmake/caches/PredefinedParams.cmake
cmake --build builds
cd ..

cd glslang
cmake -H. -Bbuilds -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX='builds/install'
cd builds
make -j8 install
cd ../../

**Reply by Flakebi** [reply]
Ok, weird. Which version of amdvlk are you trying to compile?
Also, what happens if you run `ninja GpuRtGenerateShaders` (or make, if you used that) in the amdvlk build directory?
For me, that generates the file in the build directory in `gpurt/src/pipelines/g_internal_shaders.h`.

**Comment by B4rr3l-Rid3r** [comment]
> Ok, weird. Which version of amdvlk are you trying to compile? Also, what happens if you run `ninja GpuRtGenerateShaders` (or make, if you used that) in the amdvlk build directory? For me, that generates the file in the build directory in `gpurt/src/pipelines/g_internal_shaders.h`.

Tried with master and dev, `ninja GpuRtGenerateShaders`  also didn't work, not sure exactly which directory I should try it, but tried in AMVLK, vulkandriver and driver without success 

![Screenshot from 2023-07-27 13-17-14](https://github.com/GPUOpen-Drivers/AMDVLK/assets/95323108/36c7d77f-3a35-4a2b-b9e8-bf98ae76dbba)

The structure I got after following the instructions:
![Screenshot from 2023-07-27 13-21-53](https://github.com/GPUOpen-Drivers/AMDVLK/assets/95323108/c925c095-460a-40a8-ad0a-7674e6da84a3)

![Screenshot from 2023-07-27 13-23-25](https://github.com/GPUOpen-Drivers/AMDVLK/assets/95323108/e9aded22-7ae6-4b07-b502-85bca914c575)

![Screenshot from 2023-07-27 13-23-51](https://github.com/GPUOpen-Drivers/AMDVLK/assets/95323108/f9b9ecd7-8b07-4f15-b6f0-2c53cd5df7da)

It seems the gpurt/src/pipelines are not being generated with:  cmake -G Ninja -S xgl -B builds/Release64

 I do have a GpuRtGenerateShaders.cmake file inside drivers/gpurt/cmake and also CompileRTShaders.py insider drivers/gpurt//tools but not sure how to trigger them.

**Reply by Flakebi** [reply]
> `ninja GpuRtGenerateShaders` also didn't work, not sure exactly which directory I should try it

Can you try inside the `builds/Release64` folder and post the output here please?

It should generate the file in `builds/Release64/gpurt/src/pipelines/g_internal_shaders.h`
